linker.go

```diff
@@ -164,7 +166,7 @@ func fixupJumpsAndCalls(insns asm.Instructions) error {
 			continue
 		}
 
-		return fmt.Errorf("%s at %d: reference to missing symbol %q", ins.OpCode, i, ins.Reference)
+		return fmt.Errorf("%s at insn %d: symbol %q: %w", ins.OpCode, i, ins.Reference, errUnsatisfiedReference)
```

Comment:
I think this could lead to a somewhat confusing error message: https://github.com/cilium/ebpf/blob/1946eefb4ab804a7d40b5d0d65ddd4158be73979/prog.go#L233-L235

Use a different error?

Reply:
Done.